templates/dart/test/query_test.dart.twig

```diff
@@ -20,195 +22,243 @@ class BasicFilterQueryTest {
 void main() {
   group('basic filter tests', () {
     final tests = [
-      BasicFilterQueryTest(
+      BasicFilterQueryTest<String, List<String>>(
         description: 'with a string',
         value: 's',
-        expectedValues: ["s"],
+        expectedValues: ['s'],
       ),
-      BasicFilterQueryTest(
+      BasicFilterQueryTest<int, List<int>>(
         description: 'with an integer',
         value: 1,
         expectedValues: [1],
       ),
-      BasicFilterQueryTest(
+      BasicFilterQueryTest<double, List<double>>(
         description: 'with a double',
         value: 1.2,
         expectedValues: [1.2],
       ),
-      BasicFilterQueryTest(
+      BasicFilterQueryTest<double, List<double>>(
         description: 'with a whole number double',
         value: 1.0,
         expectedValues: [1.0],
       ),
-      BasicFilterQueryTest(
+      BasicFilterQueryTest<bool, List<bool>>(
         description: 'with a bool',
         value: false,
         expectedValues: [false],
       ),
-      BasicFilterQueryTest(
+      BasicFilterQueryTest<List<String>, List<String>>(
         description: 'with a list',
         value: ['a', 'b', 'c'],
-        expectedValues: ["a","b","c"],
+        expectedValues: ['a', 'b', 'c'],
       ),
     ];
 
     group('equal()', () {
       for (var t in tests) {
         test(t.description, () {
-          final query = Query.equal('attr', t.value).toJson();
+          final query = jsonDecode(Query.equal('attr', t.value));
           expect(query['attribute'], 'attr');
           expect(query['values'], t.expectedValues);
           expect(query['method'], 'equal');
         });
       }
+
+      test('with a list', () {
+        final query = jsonDecode(Query.equal('attr', ['a', 'b', 'c']));
+        print(query);
+        expect(query['attribute'], 'attr');
+        expect(query['values'], ['a', 'b', 'c']);
+        expect(query['method'], 'equal');
+      });
     });
 
     group('notEqual()', () {
       for (var t in tests) {
         test(t.description, () {
-          final query = Query.notEqual('attr', t.value).toJson();
+          final query = jsonDecode(Query.notEqual('attr', t.value));
           expect(query['attribute'], 'attr');
           expect(query['values'], t.expectedValues);
           expect(query['method'], 'notEqual');
         });
       }
+
+      test('with a list', () {
+        final query = jsonDecode(Query.notEqual('attr', ['a', 'b', 'c']));
+        print(query);
+        expect(query['attribute'], 'attr');
+        expect(query['values'], [['a', 'b', 'c']]); // Is there a bug here?
```

Comment:
Hmm I think the idea with this was that you'd never pass a list to this because it just doesn't make sense.
```
attr != a OR attr != b OR attr != C
```
would always be true.

That said, the implementation makes the behavior weird... @lohanidamodar, what do you think?

Reply:
I think it's the database implementation, we don't support arrays for values in few methods. @abnegate might be able to shed some light.

Reply:
Not equal should only provide [a single value](https://github.com/utopia-php/database/blob/a6c1cf81e2822e78bbfcc5b07bdd31d53db6a22d/src/Database/Query.php#L336), the problem is it's hard to represent in languages without union types. This should actually [throw an exception](https://github.com/utopia-php/database/blob/main/src/Database/Validator/Query/Filter.php#L266-L277)

Reply:
@istornz, let's leave the list test for [these](https://github.com/utopia-php/database/blob/main/src/Database/Validator/Query/Filter.php#L266-L277) methods out for now. We'll figure out tests and whatnot for these in the future.